### PR TITLE
Use member initialization lists [-Weffc++] (#270)

### PR DESCRIPTION
**Pull request by jpnurmi**

### CODE REVIEW

File: include/znc/Client.h [hunk 1]
```diff
@@ -33,8 +33,7 @@ class CClient;
 
 class CAuthBase {
 public:
-	CAuthBase(const CString& sUsername, const CString& sPassword, CZNCSock* pSock) {
-		SetLoginInfo(sUsername, sPassword, pSock);
```
Comment (DarthGandalf) [inline include/znc/Client.h]: This function is not used anywhere anymore

Comment (jpnurmi) [inline include/znc/Client.h]: yeah and overriding a virtual method that is called from the constructor probably didn't do what was the intention. :) so it's ok to remove? or is this something that modules could potentially use somehow?

**Comment by jpnurmi** [comment]
:-1: crash in tests :o

**Comment by jpnurmi** [comment]
I'll get back to this when I have time to debug what goes wrong.

**Pull request closed without merging**